src/modules/home/components/EventCarousel.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="relative rounded-2xl overflow-hidden h-40 w-120 z-20 shadow">
+    <event-carousel-slide
+      v-for="(event, slideIndex) in testEventProps"
+      :key="event.eventTitle"
+      :slideIndex="slideIndex"
+      :currentIndex="currentIndex"
+      :transitionAnimation="transitionAnimation"
+    >
+      <CardBanner class="w-full" :eventTitle="event.eventTitle" />
+    </event-carousel-slide>
+    <section class="carousel-misc absolute z-10">
+      <div
+        v-for="index in indexRange"
+        :key="index"
+        @click="changeSlide(index)"
+        class="w-1"
+        :class="{
+          'carousel-gap': index !== 3
+        }"
+      >
+        <div
+          class="carousel-dot bg-white"
+          :class="{
+            'h-3': index === currentIndex,
+            'h-2 bg-opacity-30': index !== currentIndex
+          }"
+        ></div>
+      </div>
+    </section>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref } from "vue";
+import CardBanner from "./CardBanner.vue";
+import EventCarouselSlide from "./EventCarouselSlide.vue";
+import makeRange from "@/commons/utils/range.ts";
+
+export default defineComponent({
+  name: "EventCarousel",
+  components: {
+    CardBanner,
+    EventCarouselSlide
+  },
+  setup() {
+    const currentIndex = ref(0);
+    const indexRange = makeRange(4);
+    const transitionAnimation = ref("");
+    let slideTimer: number | undefined;
+    const testEventProps = [
+      { eventTitle: "Event Title 1" },
+      { eventTitle: "Event Title 2" },
+      { eventTitle: "Event Title 3" },
+      { eventTitle: "Event Title 4" }
+    ];
+
+    function startTimer() {
+      slideTimer = setInterval(function() {
+        transitionAnimation.value = "slide-down";
+        const nextSlide =

```

Comment:
This look like hard coded, can we make it changed with testEventProps?

Reply:
?????

Reply:
Number 4 is hard coded